packages/amplify-provider-awscloudformation/src/__tests__/aws-utils/IdentityPoolService.test.ts

```diff
@@ -33,11 +33,18 @@ jest.mock('../../configuration-manager', () => {
 describe('IdentityPoolService', () => {
   it('should correctly parse arn if it contains multiple forward slashes', async () => {
     const idpService = await createIdentityPoolService({} as unknown as $TSContext, {});
-    expect(await idpService.getIdentityPoolRoles('mockIdpId')).toEqual({
+    const identityPoolRoles = await idpService.getIdentityPoolRoles('mockIdpId');
+
+    // ensure role names match regex for IAM
+    // see: https://docs.aws.amazon.com/IAM/latest/APIReference/API_Role.html
+    expect(identityPoolRoles.authRoleName).toMatch(/[\w+=,.@-]+/);
```

Comment:
nit: pull the regex to a constant such as IAM_ROLE_NAME_REGEX

Reply:
Addressed in `e560e9e`